### PR TITLE
Added `betterlisting` plugin for extended queries and listing formats.

### PR DESCRIPTION
**Pull request by nikhgupta**
This PR adds a plugin to provide more template fields, functions, and query parameters. Specifically, it allows for tabularization of list results, and querying based on more fields than default ones.

Here is an example of what I have been able to achieve using this plugin:

```
± ➲ beet ls will duration_sort-
📗 ▅      04:22      01 Blurred Lines ft. T.I. & Pharrell Willia   [2013] Robin Thicke
📗 ▅      04:07      03 Get Lucky (radio edit) ft. Pharrell Will   [2013] Daft Punk
📗 ▅      04:06      40 This Is Love ft. Eva Simons                [2012] will.i.am
📗 ▅      03:52      19 Get Like Me ft. Nicki Minaj & Pharrell W   [2013] Nelly
📗 ▄      03:40      10 Something Really Bad ft. will.i.am         [2013] Dizzee Rascal
📗 ▄      03:40      16 Sonnentanz (Sun Don't Shine) ft. Will He   [2013] Klangkarussell
📗 ▄      03:22      02 Hall of Fame ft. will.i.am                 [2012] The Script
📕 ▃      02:18      16 I Will Survive                             [2013] Leah McFall
```

The first column represents whether the items have lyrics associated with them, and the second column displays the spark bars for the track duration of that item, and so on. This is just one of the examples. A simple format may look like this:

```
± ➲ beet ls will -f '$duration %rpad{$title,50} $year $artist' duration_sort-
04:22 Blurred Lines ft. T.I. & Pharrell Williams         2013 Robin Thicke
04:07 Get Lucky (radio edit) ft. Pharrell Williams       2013 Daft Punk
04:06 This Is Love ft. Eva Simons                        2012 will.i.am
03:52 Get Like Me ft. Nicki Minaj & Pharrell Williams    2013 Nelly
03:40 Something Really Bad ft. will.i.am                 2013 Dizzee Rascal
03:40 Sonnentanz (Sun Don't Shine) ft. Will Heard        2013 Klangkarussell
03:22 Hall of Fame ft. will.i.am                         2012 The Script
02:18 I Will Survive                                     2013 Leah McFall
```

Furthermore, plugin allows querying the database for fields such as ratio of lyrics present vs total tracks, average duration of tracks in an album, total duration of track/album, number of tracks missing/available/total from an album, etc.

I plan to add more fields and functions to this plugin, if it gets an okay from you guys. Note that this is the first time I have written in Python, and hence, this PR may come with a grain of salt. Do let me know what needs to be changed, and if I understand that much Python, I will try to improve upon the PR by pushing more commits.

### CODE REVIEW

File: beetsplug/betterlisting.py [hunk 1]
```diff
@@ -0,0 +1,225 @@
+# -*- coding: utf-8 -*-
+
+import re
+from beets import ui, plugins
+from beets.library import Item, Album
+
+# Other fields can be:
+#   bitrate, bpm, lyrics,
```
Comment (sampsyo) [inline beetsplug/betterlisting.py]: Stray comment?

File: beetsplug/betterlisting.py [hunk 2]
```diff
@@ -0,0 +1,225 @@
+# -*- coding: utf-8 -*-
```
Comment (sampsyo) [inline beetsplug/betterlisting.py]: Could you please add the standard heading here, and a module docstring?

File: docs/plugins/betterlisting.rst [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+querying the database. You can use :doc:`types` and specify the following
+configuration for easiery querying::
+
+    types:
+      lyrics_sort: float
+      duration_sort: int
+      avg_duration_sort: int
+      missing: int
+      available: int
+      total: int
```
Comment (sampsyo) [inline docs/plugins/betterlisting.rst]: Would it make sense to provide these types by default in this plugin?

**Comment by sampsyo** [comment]
Wow! This looks incredibly nifty—a great toolkit for building interesting terminal output. I love this hybrid retro/contemporary trend of using emoji in the console. :smiley:

I had just a couple of tiny comments in my code review. It also looks like Travis is reporting a few remaining flake8 suggestions too. After that, let's merge this in!

Thanks for the creative contribution.

**Comment by nikhgupta** [comment]
@sampsyo Is this still a go? Haven't used `beets` in a while, but wanted to close any open PR I had sent in the past.

**Comment by sampsyo** [comment]
Sure thing! We're definitely still up for it if you are.

**Comment by jtpavlock** [comment]
closing as abandoned. @nikhgupta if you feel you want to come back and take another look, feel free to re-open 😄

**Pull request closed without merging**